bench/clblast_benchmark.cpp

```diff
@@ -0,0 +1,314 @@
+  }
+
+  BENCHMARK_FUNCTION(asum_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT vr;
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(context, size);
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, &vr, 1);
+
+      cl_event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblast::Asum<ScalarT>(size, bufr.dev(), 0, buf1.dev(), 0, 1,
+                               context._queue(), &event);
+        clWaitForEvents(1, &event);
+        clReleaseEvent(event);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(nrm2_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT vr;
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(context, size);
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, &vr, 1);
+
+      cl_event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblast::Nrm2<ScalarT>(size, bufr.dev(), 0, buf1.dev(), 0, 1,
+                               context._queue(), &event);
+        clWaitForEvents(1, &event);
+        clReleaseEvent(event);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(dot_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT vr;
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(context, size);
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf2(context, size);
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, &vr, 1);
+
+      cl_event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblast::Dot<ScalarT>(size, bufr.dev(), 0, buf1.dev(), 0, 1, buf2.dev(),
+                              0, 1, context._queue(), &event);
+        clWaitForEvents(1, &event);
+        clReleaseEvent(event);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(iamax_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      int vi;
+      MemBuffer<ScalarT, CL_MEM_WRITE_ONLY> buf1(context, size);
+      MemBuffer<int, CL_MEM_READ_ONLY> buf_i(context, &vi, 1);
+
+      cl_event event;
+      flops = benchmark<>::measure(no_reps, size * 2, [&]() {
+        clblast::Amax<ScalarT>(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                               context._queue(), &event);
+        clWaitForEvents(1, &event);
+        clReleaseEvent(event);
+      });
+    }
+    return flops;
+  }
+
+  // not supported at current release yet
+  /* BENCHMARK_FUNCTION(iamin_bench) { */
+  /*   UNPACK_PARAM; */
+  /*   double flops; */
+  /*   { */
+  /*     int vi; */
+  /*     MemBuffer<ScalarT> buf1(context, size); */
+  /*     MemBuffer<int> buf_i(context, &vi, 1); */
+
+  /*     cl_event event; */
+  /*     flops = benchmark<>::measure(no_reps, [&]() { */
+  /*       clblast::Amin<ScalarT>(size, buf_i.dev(), 0, buf1.dev(), 0, 1, */
+  /*                               context._queue(), &event); */
+  /*       clWaitForEvents(1, &event); */
+  /*       clReleaseEvent(event); */
+  /*     }); */
+  /*   } */
+  /*   return flops; */
+  /* } */
```

Comment:
I think the reason was that iamin was not a part of clblast, but with release 1.0 it should be included.